interpreter/src/objects.rs

```diff
@@ -535,7 +618,23 @@ impl Value {
                     map: Arc::from(map),
                 }))
             }
-            Expression::Ident(name) => ctx.get_variable(&***name),
+            Expression::Ident(name) => {
+                // First check if this is a variable in the context
+                let var_result = ctx.get_variable(&***name);
+                if var_result.is_ok() {
+                    // If it's a variable, use that value
+                    var_result
+                } else {
+                    // Otherwise, check if it's a type identifier
+                    if let Some(cel_type) = CelType::try_from_string(name.as_str()) {
+                        // Return the type identifier as a Type value
+                        Ok(Value::Type(cel_type))
```

Comment:
Aren't type identifiers protected keywords? If so, shouldn't we check the type identifier first, then resolve the variable?

Reply:
Yup, they totally are! But only at the "root" level, but they're fine when they're nested.

I added a test case to check for that.